### ISSUE TITLE
test log likelihood probelms

### ISSUE THREAD

**Issue by ifangcheng** [thu-ml/zhusuan#98]
I noted that test_ll (test log likelihood)  is evaluated in many  examples,:
e.g. 
in bayesian_nn.py line 116: 
log_likelihood = tf.reduce_mean(zs.log_mean_exp(log_py_xw, 0)) - \
        tf.log(std_y_train)

in vae.py line 78:
is_log_likelihood = tf.reduce_mean(
        zs.is_loglikelihood(log_joint, {'x': x},
                            {'z': [qz_samples, log_qz]}, axis=0))

while in sbn_rws.py line 148:
 test_ll = sess.run(lower_bound,
                                       feed_dict={x: test_x_batch,
                                                  n_particles: ll_samples})

my questions  are:
1. why in sbn_rws.py,  just use lower_bound to evaluate test_ll , which is much different with others
2. it seems that the log_likelihood is calculated in various forms,  what's the differences between them?
is it possible to use one clean API (e.g. something like   zs.is_loglikelihood)  to unify them?

**Comment by thjashin** [comment]
In fact `zs.is_loglikelihood` computes the same value as `zs.importance_weighted_objective`, which is an importance-sampling based estimate of the log likelihood (it's also a lower bound as long as the number of samples is finite). The only difference is the latter provides gradient estimators for the variational (proposal) parameters.

So in SBN example, we avoid computing the same value twice by reusing the tensor returned by `zs.importance_weighted_objective`.

**Comment by ifangcheng** [comment]
@thjashin  very clear! thanks!
